.github/workflows/skywalking-eyes.yml

```diff
@@ -0,0 +1,47 @@
+# Copyright contributors to the oneDAL project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Check License Header
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.ref_name }}-${{ github.event.number || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  check-license-header:
+    name: Copyright Check
+    if: github.event.repository.fork == false
+    runs-on: ubuntu-24.04
+    permissions:
+      issues: write
+      pull-requests: write
```

Comment:
Are write permissions necessary for this?

Reply:
The check can leave comments on the pull request if it finds issues. For this, write permissions need to be granted for these scopes.